### ISSUE TITLE
`match-component-import-name`: ignore if there is a conflicting type definition

### ISSUE THREAD

**Issue by FloEdelmann** [vuejs/eslint-plugin-vue#2011]
**What rule do you want to change?** [`vue/match-component-import-name`](https://eslint.vuejs.org/rules/match-component-import-name.html)

We often have components with the same name as TypeScript interfaces. Thus we have to rename one of them. We usually append `Component` to the imported component, but register it without this suffix:

```ts
import type { FooBar } from './types'
import FooBarComponent from './FooBar.vue'

export default defineComponent({
  components: {
    FooBar: FooBarComponent,
  },
})
```

Of course, the `vue/match-component-import-name` rule complains that the imported name is not equal to the used name. But using `<FooBarComponent>` in the template seems weird, and renaming to `FooBar` is not possible because that name is already taken.

**Does this change cause the rule to produce more or fewer warnings?**
Fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**
New option `ignoreConflictingNames: true` will not report `FooBar: Anything` if the name `FooBar` is already taken by a variable or type. Default should be `ignoreConflictingNames: false`, i.e. the current behavior.
